### PR TITLE
Refactor/dynamic objects lists

### PR DESCRIPTION
**Pull request by metelik**
- CSV objects created now dynamically
- Removing initialisation code that was relying on a pre-compiled set maximum no. of objects

### CODE REVIEW

File: src/bacnet/basic/object/csv.c [hunk 1]
```diff
@@ -80,29 +78,70 @@ void CharacterString_Value_Property_Lists(
     return;
 }
 
+/**
+ * @brief Creates a CharacterString Value object
+ * @param object_instance - object-instance number of the object
+ * @return the object-instance that was created, or BACNET_MAX_INSTANCE
+ */
+uint32_t CharacterString_Value_Create(uint32_t object_instance)
```
Comment (adrigonzo) [inline src/bacnet/basic/object/csv.c:86]: are there more _Create() functions coming for the other object types?

File: src/bacnet/basic/object/csv.c [hunk 2]
```diff
@@ -439,22 +423,18 @@ static char *CharacterString_Value_Description(uint32_t object_instance)
  * @return True on success, false otherwise.
  */
 bool CharacterString_Value_Description_Set(
-    uint32_t object_instance, BACNET_CHARACTER_STRING *new_descr)
+    uint32_t object_instance, const char *new_descr)
```
Comment (adrigonzo) [inline src/bacnet/basic/object/csv.c:426]: `const char *const` ?? 😋

**Review by adrigonzo — APPROVED** [review]

File: src/bacnet/basic/object/csv.c (hunk 1 above)
Comment (metelik) [reply src/bacnet/basic/object/csv.c:86]: There are... This one had not been yet updated by the upstream.

File: src/bacnet/basic/object/csv.c (hunk 2 above)
Comment (metelik) [reply src/bacnet/basic/object/csv.c:426]: Lets be bold const char * const restrict new_descr; ;)

**Pull request merged**